accessibility_api/AccessibilityRequestHandler.py

```diff
@@ -0,0 +1,127 @@
+'''
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this file,
+You can obtain one at http://mozilla.org/MPL/2.0/.
+'''
+
+import sys
+from BaseHTTPServer import BaseHTTPRequestHandler
+from urlparse import urlparse, parse_qs, parse_qsl
+from json import dumps
+from comtypes import CoInitialize
+from accessibility_api.accessibility_lib.scripts.constants import (
+    SUCCESSFUL_RESPONSE, ERROR_RESPONSE
+)
+from accessibility_api.accessibility_lib.scripts.accessible import accessible
+from accessibility_api.accessibility_lib.scripts.event import event
+from accessibility_api.accessibility_lib.scripts.commands import (
+    execute_command
+)
+
+
+class AccessibilityRequestHandler(BaseHTTPRequestHandler):
+    """ Service requests for accessible objects """
+    def _successful_response(self, json):
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/json')
+        self.end_headers()
+        self.wfile.write(json)
+
+    def _bad_request(self, message):
+        self.send_response(500, message)
+        self.send_header('Content-Type', 'application/json')
+        self.end_headers()
+
+    def do_GET(self):
+        '''Proccess get requests'''
+        url = urlparse(self.path)
+        handler = getattr(self, "get_" + url.path[1:], None)
+        if handler:
+            handler(url.query)
+        else:
+            self.send_response(403, 'Invalid Request')
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+
+        return
+
+    def response(self, message, key, value, status):
+        """Returns appropriate response"""
+        response_args = {
+            ERROR_RESPONSE: message,
+            SUCCESSFUL_RESPONSE: dumps({key: value})
+        }[status]
+
+        return {
+            ERROR_RESPONSE: self._bad_request,
+            SUCCESSFUL_RESPONSE: self._successful_response
+        }[status](response_args)
+
+    def get_accessible(self, urlquery):
+        '''Retrieve accessible object'''
+
+        params = dict(parse_qsl(urlquery))
+        if params.get('interface') is None:
+            self._bad_request('Bad Request: No interface specified')
+            return
+
+        acc_obj = accessible(params).serialize_result(params.get('depth'))
+        self.response(
+            'Bad Request: No accessible exists',
+            params.get('interface'),
```

Comment:
so this argument (key) here and in other get_... functions should be encapsulated inside the accessible(), event(), etc. Ideally your object should have result that is already, is serialized and corresponds to the json value that you 'get' in the self.response call.

Once you do that, you will notice that your methods are going to be pretty much the same, with the exception of the check for a required parameter (interface, type, function) and the type of the class that you create. See if you can abstract that even more into 1 function :)